src/OSPSuite.Core/Domain/Formulas/DistributionFormulaFactory.cs

```diff
@@ -21,68 +25,85 @@ public interface IDistributionFormulaFactory
-         addObjectPathToDistributionFormula(distribution, distributedParameter,deviationParameter,Constants.Distribution.DEVIATION);
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, deviationParameterName, DEVIATION, dimension);
+      };
 
-      public DiscreteDistributionFormula CreateDiscreteDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter)
+      public DiscreteDistributionFormula CreateDiscreteDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter) =>
+         createDistribution<DiscreteDistributionFormula>(meanParameter.Dimension, configureDiscreteDistribution(meanParameter.Name));
+
+      private Action<DistributionFormula, IDimension> configureDiscreteDistribution(string meanParameterName = null) => (distributionFormula, dimension) =>
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+
+      public LogNormalDistributionFormula CreateLogNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter) =>
+         createDistribution<LogNormalDistributionFormula>(meanParameter.Dimension, configureLogNormalDistributionDistribution(meanParameter.Name, deviationParameter.Name));
+
+      private Action<DistributionFormula, IDimension> configureLogNormalDistributionDistribution(string meanParameterName = null, string deviationParameter = null) => (distributionFormula, dimension) =>
       {
-         var distribution = createDistribution<DiscreteDistributionFormula>();
-         distribution.Dimension = meanParameter.Dimension;
-         addObjectPathToDistributionFormula(distribution, distributedParameter,meanParameter,Constants.Distribution.MEAN);
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, deviationParameter, GEOMETRIC_DEVIATION, null);
+      };
+
+      public UniformDistributionFormula CreateUniformDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter minParameter, IParameter maxParameter) =>
+         createDistribution<UniformDistributionFormula>(minParameter.Dimension, configureUniformDistributionDistribution(minParameter.Name, maxParameter.Name));
 
-      public LogNormalDistributionFormula CreateLogNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter)
+      private Action<DistributionFormula, IDimension> configureUniformDistributionDistribution(string minParameterName = null, string maxParameterName = null) => (distributionFormula, dimension) =>
```

Comment:
same (null)

Reply:
rename **configureUniformDistributionDistribution** to **configureUniformDistribution**